modules/notification/ui/ui.go

```diff
@@ -36,51 +37,59 @@ func NewNotifier() base.Notifier {
 
 func (ns *notificationService) Run() {
 	for opts := range ns.issueQueue {
-		if err := models.CreateOrUpdateIssueNotifications(opts.issue, opts.notificationAuthorID); err != nil {
+		if err := models.CreateOrUpdateIssueNotifications(opts.issueID, opts.commentID, opts.notificationAuthorID); err != nil {
 			log.Error("Was unable to create issue notification: %v", err)
 		}
 	}
 }
 
 func (ns *notificationService) NotifyCreateIssueComment(doer *models.User, repo *models.Repository,
 	issue *models.Issue, comment *models.Comment) {
-	ns.issueQueue <- issueNotificationOpts{
-		issue,
-		doer.ID,
+	var opts = issueNotificationOpts{
+		issueID:              issue.ID,
+		notificationAuthorID: doer.ID,
+	}
+	if comment != nil {
```

Comment:
Question: why would comment be nil here? Is this for compatibility?

Reply:
A comment maybe deleted before you read the notification.